kernel/constant_typing.ml

```diff
@@ -28,62 +27,39 @@ module NamedDecl = Context.Named.Declaration
 (* Checks the section variables for the body.
    Returns the closure of the union with the variables in the type.
 *)
-let check_section_variables env declared_set typ body =
+let check_section_variables env declared_vars body typ =
+  let env_ids = ids_of_named_context_val (named_context_val env) in
+  Id.Set.iter (fun id -> if not (Id.Set.mem id env_ids) then Type_errors.error_unbound_var env id) declared_vars;
```

Comment:
Should this error be an anomaly or is it accessible with a bad `using` annotation?

Reply:
> Should this error be an anomaly

> The checker should be robust against this

Indeed, morally this is not supposed to happen, and indeed, somehow, all kernel errors are supposed to be anticipated. So, ok to leave as it.

> is it accessible with a bad using annotation?

Afaiu, it is supposed to be checked vernac-side, but there is a hole with the `#[using=..., program]` attribute.

The test-suite already spotted a problem with `#[using="All", program]` but I just find another one which silently ignores an invalid name in master (and is caught only in the kernel with the PR before the new commit I just added):
```
Require Import Program.
#[using="dummy", program]
Fixpoint f (x:nat) : nat := match x with 0 => 0 | S n => S n end.
```